runtime/src/main/cpp/Memory.h

```diff
@@ -356,23 +357,16 @@ class ObjHolder {
    ObjHeader* obj_;
 };
 
-//! TODO Follow the Rule of Zero to prevent dangling on unintented copy ctor
 class ExceptionObjHolder {
- public:
-   explicit ExceptionObjHolder(const ObjHeader* obj) {
-     ::SetHeapRef(&obj_, obj);
-   }
-
-   ~ExceptionObjHolder() {
-     ZeroHeapRef(&obj_);
-   }
-
-   ObjHeader* obj() { return obj_; }
+public:
+#if !KONAN_NO_EXCEPTIONS
+    static void Throw(ObjHeader* exception) RUNTIME_NORETURN;
+#endif
 
-   const ObjHeader* obj() const { return obj_; }
+    // Exceptions are not on a hot path, so having virtual dispatch is fine.
```

Comment:
Can additional code on cold path affect code generation of hot path?

Reply:
I don't think it should: LLVM is well aware that the additional code lives in the cold section (as we're explicitly adding it to the landingpad section). I will, however, check with our benchmarks.

Reply:
Ok, so, using virtual dispatch caused some regressions and some improvements. It probably has something to do with layout again, but it's not certain. So I rewrote it to static dispatch with `ALWAYS_INLINE`, which normalized the situation.

Reply:
Just curious: what's the point for `GetExceptionObject` being static?

Reply:
No point whatsoever, fixed.